libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/internal/kapt/Kapt3KotlinGradleSubplugin.kt

```diff
@@ -428,14 +428,18 @@ class Kapt3KotlinGradleSubplugin : KotlinGradleSubplugin<KotlinCompile> {
             kaptTask.annotationProcessorOptionProviders.add(it)
         }
 
+        val dslJavacOptions = kaptExtension.getJavacOptions().toMutableMap()
+        if (javaCompile != null && "-source" !in dslJavacOptions) {
```

Comment:
Probably need to check for `--release N` too which combines `-source N`, `-target N` and `-bootclasspath <from-N>` http://openjdk.java.net/jeps/247

Reply:
Good catch, I was not aware of this option.